tests/simpleKVBC/TesterReplica/main.cpp

```diff
@@ -54,17 +56,25 @@ void run_replica(int argc, char** argv) {
   // registration of metrics from the replica with the aggregator and don't
   // return empty metrics from the metrics server.
   setup->GetMetricsServer().Start();
-  while (replica->isRunning()) std::this_thread::sleep_for(std::chrono::seconds(1));
+  while (replica->isRunning()) {
+    if (timeToExit) {
+      setup->GetMetricsServer().Stop();
+      replica->stop();
+    } else {
+      std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+  }
 }
 }  // namespace concord::kvbc::test
 
 using namespace std;
 
-void signal_handler(int signal_num) {
+namespace {
+static void signal_handler(int signal_num) {
   LOG_INFO(GL, "Program received signal " << signal_num);
-  exit(0);
+  concord::kvbc::test::timeToExit = true;
```

Comment:
Nice, that is clean solution!

Nit: I don't want to be a language lawyer here, especially when we log in signal handlers via global logger objects, but if the atomic variable  is not lock-free, that won't be safe as it implies a mutex that can be held by the same thread running the handler. That obviously matters mainly for portability.

Not sure if we should handle that as we have C-style signal handling all over the place. Might be a good idea to still check if the variable is lock-free or don't use `signal()` for signal-handling. We can use `sigaction` instead. Or better yet - boost.

Ref: https://en.cppreference.com/w/cpp/atomic/atomic_is_lock_free